### PR TITLE
HARP-10717: Restrict camera movements in MapControls to stay within g…

### PR DESCRIPTION
**Pull request by atomicsulfate**
…iven geo bounds.

Signed-off-by: Andres Mandado <andres.mandado-almajano@here.com>

Thank you for contributing to harp.gl!

Before requesting a pull request, please remember to check the following documents:
* [contribution guidelines](https://github.com/heremaps/harp.gl/blob/master/CONTRIBUTING.md)
* [coding style](https://github.com/heremaps/harp.gl/blob/master/CODINGSTYLE.md)

If you are adding new functionality we would highly appreciate if you can describe what is the capability you are adding and even better if you can add some examples. Please also remember to add tests for it.

# CI Check

Our bots will check whether your PR can be directly integrated into the mainline. We have some internal integration tests running on the background, our bots will inform you of the next steps and someone from our team will take a look and help if needed!

And please do not forget to sign-off your commit! You can read more about DCO [here](https://julien.ponge.org/blog/developer-certificate-of-origin-versus-contributor-license-agreements/). But, in short, you just need to use `git commit -s` or append `--signoff` when you are committing to the repo.

Happy contributing!

### CODE REVIEW

**Comment by atomicsulfate** [comment]
@heremaps-bot retest this please

**Comment by atomicsulfate** [comment]
@harpgl-bot retest this please

File: @here/harp-map-controls/lib/MapControls.ts [hunk 1]
```diff
@@ -143,6 +143,12 @@ export class MapControls extends THREE.EventDispatcher {
      */
     rotateEnabled = true;
 
+    /**
+     * If defined, restricts all camera movements (pan, zoom, orbit) triggered from these map
+     * controls to the specified bounding box.
+     */
+    bounds?: geoUtils.GeoBox;
```
Comment (ninok) [inline @here/harp-map-controls/lib/MapControls.ts:150]: The documentation is a bit imprecise. Do we restrict the target point to be in the bounds or the camera or the whole view frustum?

Comment (atomicsulfate) [reply @here/harp-map-controls/lib/MapControls.ts:150]: I'll clarify that, it's the camera position.

File: @here/harp-map-controls/lib/MapControls.ts [hunk 2]
```diff
@@ -147,7 +147,7 @@ export class MapControls extends THREE.EventDispatcher {
      * If defined, restricts all camera movements (pan, zoom, orbit) triggered from these map
      * controls so that camera position remains within the specified bounding box.
      */
-    bounds?: geoUtils.GeoBox;
+    bounds?: GeoBox;
```
Comment (atomicsulfate) [inline @here/harp-map-controls/lib/MapControls.ts:150]: this was the offending line for api-extractor. It was the first time a MapControls member (appearing in MapControls.d.ts) was using a type imported with wildcard.

File: @here/harp-map-controls/lib/MapControls.ts (hunk 1 above)
Comment (ninok) [reply @here/harp-map-controls/lib/MapControls.ts]: Okay, but maybe we need to clarify with the customer what they actually want. Especially for the usecase where you only want to see "one" world you need to restrict the frustum, not only the camera position.

Comment (atomicsulfate) [reply @here/harp-map-controls/lib/MapControls.ts]: I can double check. But the ticket is quite clear they want to restrict movement, not visibility. 
Quoting https://saeljira.it.here.com/browse/HARP-10717 :
"restrict the camera movements to a particular bounds"

MapBox has also the same functionality.

**Pull request closed without merging**